### PR TITLE
Add sql, reporting, observability, gantt chart for 1.2

### PR DESCRIPTION
**Pull request by joshuali925**
### Description
- add sql, reporting, observability (in trace-analytics repo), and gantt-chart
- remove duplicated `schema-version: '1.0'`
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dblock** [comment]
I'm fixing CI for the 1.1.0 manifest.

File: manifests/1.2.0/opensearch-1.2.0.yml [hunk 1]
```diff
@@ -64,3 +64,23 @@ components:
...
+    checks:
+      - gradle:properties:version
+      - gradle:dependencies:opensearch.version
+  - name: opensearch-observability
+    repository: https://github.com/opensearch-project/trace-analytics.git
+    ref: "main"
+    working_directory: "opensearch-observability"
+    checks:
+      - gradle:properties:version
+      - gradle:dependencies:opensearch.version
```
Comment (gaiksaya) [inline manifests/1.2.0/opensearch-1.2.0.yml:86]: Looks like we have new plugin. Adding @peternied as he is looking over 1.2 release.

**Comment by kavilla** [comment]
Unfortunately since this is a new(ish) plugin and it's in a "hybrid" repo it can't use the `default` script for building OpenSearch Dashboards plugins. Could you add a `observabilityDashboards/build.sh` to here https://github.com/opensearch-project/opensearch-build/tree/main/scripts/components. With the `build.sh` being:

```
#!/bin/bash

# SPDX-License-Identifier: Apache-2.0
#
# The OpenSearch Contributors require contributions made to
# this file be licensed under the Apache-2.0 license or a
# compatible open source license.

set -ex

function usage() {
    echo "Usage: $0 [args]"
    echo ""
    echo "Arguments:"
    echo -e "-v VERSION\t[Required] OpenSearch version."
    echo -e "-s SNAPSHOT\t[Optional] Build a snapshot, default is 'false'."
    echo -e "-p PLATFORM\t[Optional] Platform, ignored."
    echo -e "-a ARCHITECTURE\t[Optional] Build architecture, ignored."
    echo -e "-o OUTPUT\t[Optional] Output path, default is 'artifacts'."
    echo -e "-h help"
}

while getopts ":h:v:s:o:p:a:" arg; do
    case $arg in
        h)
            usage
            exit 1
            ;;
        v)
            VERSION=$OPTARG
            ;;
        s)
            SNAPSHOT=$OPTARG
            ;;
        o)
            OUTPUT=$OPTARG
            ;;
        p)
            PLATFORM=$OPTARG
            ;;
        a)
            ARCHITECTURE=$OPTARG
            ;;
        :)
            echo "Error: -${OPTARG} requires an argument"
            usage
            exit 1
            ;;
        ?)
            echo "Invalid option: -${arg}"
            exit 1
            ;;
    esac
done

if [ -z "$VERSION" ]; then
    echo "Error: You must specify the OpenSearch Dashboards version"
    usage
    exit 1
fi

[ -z "$OUTPUT" ] && OUTPUT=artifacts

mkdir -p $OUTPUT/plugins
# For hybrid plugin it actually resides in 'trace-analytics/dashboards-observability'
PLUGIN_FOLDER=$(basename "$PWD")
PLUGIN_NAME=$(basename $(dirname "$PWD"))
# TODO: [CLEANUP] Needed OpenSearch Dashboards git repo to build the required modules for plugins
# This makes it so there is a dependency on having Dashboards pulled already.
cp -r ../$PLUGIN_FOLDER/ ../../OpenSearch-Dashboards/plugins
echo "BUILD MODULES FOR $PLUGIN_NAME"
(cd ../../OpenSearch-Dashboards && yarn osd bootstrap)
echo "BUILD RELEASE ZIP FOR $PLUGIN_NAME"
(cd ../../OpenSearch-Dashboards/plugins/$PLUGIN_FOLDER && yarn plugin_helpers build)
echo "COPY $PLUGIN_NAME.zip"
cp -r ../../OpenSearch-Dashboards/plugins/$PLUGIN_FOLDER/build/$PLUGIN_NAME-$VERSION.zip $OUTPUT/plugins/
rm -rf ../../OpenSearch-Dashboards/plugins/$PLUGIN_FOLDER
```

After adding that file, I was able to build it.  But, I can do it for you if you don't mind me committing on top of your branch.

Will need someone from OpenSearch to verify the build since I'm having issues with kNN locally.

**Review by kavilla — CHANGES_REQUESTED** [review]
Needs a new build script added for the new component.

File: manifests/1.2.0/opensearch-1.2.0.yml (hunk 1 above)
Comment (peternied) [reply manifests/1.2.0/opensearch-1.2.0.yml:86]: Thanks for watching out for this, the repo for trace analytics has been included in the release process as https://github.com/opensearch-project/trace-analytics/issues/118.  To what is included or not is up to the repo to manage as they are onboard the release process 👍 .

**Review by dblock — APPROVED** [review]

File: manifests/1.2.0/opensearch-1.2.0.yml [hunk 2]
```diff
@@ -64,15 +64,23 @@ components:
     platforms:
       - darwin
       - linux
-  - name: anomaly-detection
```
Comment (dblock) [inline manifests/1.2.0/opensearch-1.2.0.yml:67]: I don't think you meant to remove AD and CCR

**Review by dblock — CHANGES_REQUESTED** [review]

File: manifests/1.2.0/opensearch-1.2.0.yml (hunk 2 above)
Comment (anirudha) [reply manifests/1.2.0/opensearch-1.2.0.yml]: yes; seem to be an accident, we should not remove other plugins that we don't own from here.

Comment (joshuali925) [reply manifests/1.2.0/opensearch-1.2.0.yml]: accidentally removed during rebase, updated

**Review by anirudha — APPROVED** [review]
lgtm

**Comment by anirudha** [comment]
@dblock @kavilla could you please review and approve , thanks

**Review by dblock — APPROVED** [review]

**Review by kavilla — APPROVED** [review]
Works for me on Dashboards side.

**Pull request merged**